File: atcoder/abc324a_same.cpp
// Vicfred
// https://atcoder.jp/contests/abc324/tasks/abc324_a
// implementation
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int N;
  cin >> N;
  vector<int> A(N);
  for(int i = 0; i < N; ++i) {
    cin >> A[i];
  }
  int same = A[0];
  for(int i = 1; i < N; ++i) {
    if(A[i] != same) {
      cout << "No" << endl;
      return 0;
    }
  }
  cout << "Yes" << endl;
  return 0;
}

File: atcoder/abc324a_same.hs
-- vicfred
-- https://atcoder.jp/contests/abc324/tasks/abc324_a
-- implementation
import Data.List (group)
import Data.Bool (bool)

main :: IO ()
main = getContents >>= putStrLn .bool "No" "Yes".(== 1)
                                .length
                                .group
                                .words
                                .last
                                .lines
